plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -748,27 +800,32 @@ fu_genesys_usbhub_device_setup(FuDevice *device, GError **error)
 				   (guint8 *)&self->static_ts,
 				   sizeof(FuGenesysStaticToolString));
 	}
-	if (self->static_ts.tool_string_version != 0xff) {
-		gchar rev[3] = {0};
+	if (memcmp(&self->static_ts.tool_string_version, "USB", 3) == 0) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INTERNAL,
+			    "Unsupported IC: tool string unsupported");
+		return FALSE;
+	}
 
-		if (memcmp(self->static_ts.mask_project_ic_type, "3523", 4) == 0) {
-			self->isp_model = ISP_MODEL_HUB_GL3523;
-		} else if (memcmp(self->static_ts.mask_project_ic_type, "3590", 4) == 0) {
-			self->isp_model = ISP_MODEL_HUB_GL3590;
-		} else {
-			g_autofree gchar *ic_type =
-			    fu_common_strsafe((const gchar *)&self->static_ts.mask_project_ic_type,
-					      sizeof(self->static_ts.mask_project_ic_type));
-			g_set_error(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_NOT_SUPPORTED,
-				    "IC type %s not supported",
-				    ic_type);
-			return FALSE;
-		}
-		memcpy(rev, &self->static_ts.mask_project_ic_type[4], 2);
-		self->isp_revision = fu_common_strtoull(rev);
+	if (memcmp(self->static_ts.mask_project_ic_type, "3523", 4) == 0) {
+		self->chip.model = ISP_MODEL_HUB_GL3523;
+	} else if (memcmp(self->static_ts.mask_project_ic_type, "3590", 4) == 0) {
+		self->chip.model = ISP_MODEL_HUB_GL3590;
+	} else {
+		g_autofree gchar *ic_type =
+		    fu_common_strsafe((const gchar *)&self->static_ts.mask_project_ic_type,
+				      sizeof(self->static_ts.mask_project_ic_type));
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "IC type %s not supported",
+			    ic_type);
+		return FALSE;
 	}
+	memcpy(rev, &self->static_ts.mask_project_ic_type[4], 2);
+	tmp = fu_common_strtoull(rev);
+	self->chip.revision = tmp;
```

Comment:
You can get rid of the `tmp` variable with `self->chip.revision = fu_common_strtoull(rev);` as it was changed by the commit 79228d8b7d69dd76277a85e444835e38aaf6b954.